src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -81,74 +94,114 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
-    mean_sim_body_weight /= population_size;
-    mean_adjustment_coefficient /= population_size;
-    double shift = (target_BW - mean_sim_body_weight) / mean_adjustment_coefficient;
+    // Intervention scenario: recieve adjustments from baseline scenario.
+    else {
+        auto message = context.scenario().channel().try_receive(context.sync_timeout_millis());
+        if (!message.has_value()) {
+            throw core::HgpsException(
+                "Simulation out of sync, receive Kevin Hall adjustments message has timed out");
+        }
+
+        auto &basePtr = message.value();
+        auto *messagePrt = dynamic_cast<KevinHallAdjustmentMessage *>(basePtr.get());
+        if (!messagePrt) {
+            throw core::HgpsException(
+                "Simulation out of sync, failed to receive a Kevin Hall adjustments message");
+        }
 
-    // Final run.
+        adjustments = messagePrt->data();
+    }
+
+    // Adjust: different model for children and adults.
     for (auto &person : context.population()) {
         // Ignore if inactive.
         if (!person.is_active()) {
             continue;
         }
 
-        // TODO: Simulate person and update risk factors.
-        simulate_person(person, shift);
-        // SimulatePersonState state = simulate_person(person, shift);
-        // person.risk_factors["Height"_id] = state.H;
-        // person.risk_factors["Weight"_id] = state.BW;
-        // person.risk_factors["PhysicalActivity"_id] = state.PAL;
-        // person.risk_factors["RestingMetabolicRate"_id] = state.RMR;
-        // person.risk_factors["BodyFat"_id] = state.F;
-        // person.risk_factors["LeanTissue"_id] = state.L;
-        // person.risk_factors["ExtracellularFluid"_id] = state.ECF;
-        // person.risk_factors["Glycogen"_id] = state.G;
-        // person.risk_factors["EnergyExpenditure"_id] = state.EE;
-        // person.risk_factors["EnergyIntake"_id] = state.EI;
+        auto &adjustment = adjustments.at(person.gender, person.age);
+
+        if (person.age < kevin_hall_age_min) {
+            initialise_kevin_hall_state(person, adjustment);
+        } else {
+            kevin_hall_adjust(person, adjustment);
+        }
+    }
+
+    // Baseline scenario: send adjustments to intervention scenario.
+    if (context.scenario().type() == ScenarioType::baseline) {
+        context.scenario().channel().send(std::make_unique<KevinHallAdjustmentMessage>(
+            context.current_run(), context.time_now(), std::move(adjustments)));
     }
 }
 
+KevinHallAdjustmentTable
+KevinHallModel::compute_kevin_hall_adjustments(Population &population) const {
+    auto expected = get_risk_factor_expected();
+    auto weight_means = compute_mean_weight(population);
+
+    // Compute adjustments.
+    auto adjustments = KevinHallAdjustmentTable{};
+    for (const auto &[sex, weight_means_by_sex] : weight_means) {
+        adjustments.emplace_row(sex, std::unordered_map<int, double>{});
```

Comment:
I think this will work:

```suggestion
        adjustments.emplace_row(sex, {});
```